src/codegen/spirv/ir_builder.cc

```diff
@@ -552,48 +552,45 @@ Value IRBuilder::Mod(Value a, Value b) {
   }
 }
 
-
-#define DEFINE_BUILDER_CMP_OP(_OpName, _Op)                        \
-  Value IRBuilder:: _OpName(Value a, Value b) {                    \
-    CHECK_EQ(a.stype.id, b.stype.id);                              \
-    if (t_bool_.id == 0) {                                         \
-      t_bool_ = DeclareType(UInt(1));                              \
-    }                                                              \
-    if (a.stype.type.is_int()) {                                   \
-      return MakeValue(spv::OpS ## _Op, t_bool_, a, b);            \
-    } else if (a.stype.type.is_uint()) {                           \
-      return MakeValue(spv::OpU ## _Op, t_bool_, a, b);            \
-    } else {                                                       \
-      CHECK(a.stype.type.is_float());                              \
-      return MakeValue(spv::OpFOrd ## _Op, t_bool_, a, b);         \
-    }                                                              \
+#define DEFINE_BUILDER_CMP_OP(_OpName, _Op)                                           \
+  Value IRBuilder::_OpName(Value a, Value b) {                                        \
+    CHECK_EQ(a.stype.id, b.stype.id);                                                 \
+    CHECK_EQ(a.stype.type.lanes(), b.stype.type.lanes());                             \
+    const auto& bool_type = this->GetSType(UInt(1).with_lanes(a.stype.type.lanes())); \
+    if (a.stype.type.is_int()) {                                                      \
+      return MakeValue(spv::OpS##_Op, bool_type, a, b);                               \
```

Comment:
For consistency, it'd be good to keep the the spaces separating '##' as it is in the binary builders.